From 25a3a1adb2fe59a9cf886417c4c6266ed17e7329 Mon Sep 17 00:00:00 2001
From: Oliver Brown <oliver.brown@nxp.com>
Date: Mon, 16 Jul 2018 08:07:28 -0500
Subject: [PATCH 4324/5242] MLK-18904-2: imx: hdp: Moving common functions,
 making private functions static

commit  8d49f6a1b3df78bb92287b8f51857f885fe77f9c from
https://source.codeaurora.org/external/imx/linux-imx.git

Moving AFE_check_rate_supported to API_AFE.c (another commit).
Making private functions static to prevent conflicts with iMX8M (MCU2).

Signed-off-by: Oliver Brown <oliver.brown@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/imx/hdp/API_AFE_mcu1_dp.c |   28 +++++++---------------------
 drivers/gpu/drm/imx/hdp/API_AFE_mcu1_dp.h |    4 ++--
 2 files changed, 9 insertions(+), 23 deletions(-)

diff --git a/drivers/gpu/drm/imx/hdp/API_AFE_mcu1_dp.c b/drivers/gpu/drm/imx/hdp/API_AFE_mcu1_dp.c
index be26590..c228914 100644
--- a/drivers/gpu/drm/imx/hdp/API_AFE_mcu1_dp.c
+++ b/drivers/gpu/drm/imx/hdp/API_AFE_mcu1_dp.c
@@ -48,22 +48,6 @@
 #include "API_AFE_mcu1_dp.h"
 #include "../../../../mxc/hdp/all.h"
 
-u8 AFE_check_rate_supported(ENUM_AFE_LINK_RATE rate)
-{
-	switch (rate) {
-	case AFE_LINK_RATE_1_6:
-	case AFE_LINK_RATE_2_1:
-	case AFE_LINK_RATE_2_4:
-	case AFE_LINK_RATE_2_7:
-	case AFE_LINK_RATE_3_2:
-	case AFE_LINK_RATE_4_3:
-	case AFE_LINK_RATE_5_4:
-		return 1;
-	default:
-		return 0;
-	}
-}
-
 static void AFE_WriteReg(state_struct *state, ENUM_AFE_LINK_RATE link_rate,
 			 unsigned int addr,
 			 unsigned int val1_6,
@@ -103,7 +87,7 @@ static void AFE_WriteReg(state_struct *state, ENUM_AFE_LINK_RATE link_rate,
 	}
 }
 
-void phy_cfg_24mhz(state_struct *state, int num_lanes)
+static void phy_cfg_24mhz(state_struct *state, int num_lanes)
 {
 	int k;
 
@@ -115,7 +99,9 @@ void phy_cfg_24mhz(state_struct *state, int num_lanes)
 }
 
 /* Valid for 24 MHz only */
-void phy_cfg_dp_pll0(state_struct *state, int num_lanes, ENUM_AFE_LINK_RATE link_rate)
+static void phy_cfg_dp_pll0(state_struct *state,
+			    int num_lanes,
+			    ENUM_AFE_LINK_RATE link_rate)
 {
 	int k;
 	volatile u16 rdata;
@@ -236,7 +222,7 @@ void phy_cfg_dp_pll0(state_struct *state, int num_lanes, ENUM_AFE_LINK_RATE link
 	}
 }
 
-void phy_cfg_dp_ln(state_struct *state, int num_lanes)
+static void phy_cfg_dp_ln(state_struct *state, int num_lanes)
 {
 	int k;
 	u16 rdata;
@@ -261,7 +247,7 @@ void phy_cfg_dp_ln(state_struct *state, int num_lanes)
 	}
 }
 
-u16 aux_cal_cfg(state_struct *state, u16 prev_calib_code)
+static u16 aux_cal_cfg(state_struct *state, u16 prev_calib_code)
 {
 	u16 txpu_calib_code;
 	u16 txpd_calib_code;
@@ -292,7 +278,7 @@ u16 aux_cal_cfg(state_struct *state, u16 prev_calib_code)
 	return new_calib_code;
 }
 
-void aux_cfg(state_struct *state)
+static void aux_cfg(state_struct *state)
 {
 	volatile u16 rdata;
 
diff --git a/drivers/gpu/drm/imx/hdp/API_AFE_mcu1_dp.h b/drivers/gpu/drm/imx/hdp/API_AFE_mcu1_dp.h
index 1813c4f..af71347 100644
--- a/drivers/gpu/drm/imx/hdp/API_AFE_mcu1_dp.h
+++ b/drivers/gpu/drm/imx/hdp/API_AFE_mcu1_dp.h
@@ -128,7 +128,7 @@
 #define TX_DIAG_TX_DRV_1 0x43E1
 #define TX_DIAG_TX_DRV_2 0x45E1
 #define TX_DIAG_TX_DRV_3 0x47E1
-#define PHY_HDP_MODE CTRL 0xC008
+#define PHY_HDP_MODE_CTRL 0xC008
 #define TX_DIAG_ACYA_0 0x41ff
 #define TX_DIAG_ACYA_1 0x43ff
 #define TX_DIAG_ACYA_2 0x45ff
@@ -165,4 +165,4 @@
 #define PHY_HDP_TX_CTL_L2 0xC488
 #define PHY_HDP_TX_CTL_L3 0xC4C8
 
-#endif //API_AFE_MCU1_DP_H
+#endif /*API_AFE_MCU1_DP_H*/
-- 
1.7.9.5

